photon-ml/src/main/scala/com/linkedin/photon/ml/data/ScoredGameDatum.scala

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.data
+
+import scala.collection.Map
+
+import com.linkedin.photon.ml.util.Utils.isAlmostZero
+
+/**
+  * Representation of a single scored GAME data point
```

Comment:
@fastier-li `GameDatum` is a beefy data structure since it contains the feature vectors for all shards. This is impractical to pass around for shuffles. While we can be careful with tricks like co-partitioning, this would be a contract contributors will have to keep track of which leads to maintenance risks. Even if we enforced that, co-partitioning does not ensure co-location and passing around GameData over network traffic is not optimal either.

I am also working on a document that explains what we tried and how we got from unable to score certain datasets to not meeting required SLAs to the current solution. I will also add some of the relevant information as documentation in the code.